gr-digital/lib/map_bb_impl.cc

```diff
@@ -49,18 +49,19 @@ void map_bb_impl::set_map(const std::vector<int>& map)
 {
     gr::thread::scoped_lock guard(d_mutex);
 
-    for (int i = 0; i < 0x100; i++)
+    for (int i = 0; i < d_map_size; i++)
         d_map[i] = i;
 
-    unsigned int size = std::min((size_t)0x100, map.size());
+    const unsigned int size = std::min(d_map_size, map.size());
     for (unsigned int i = 0; i < size; i++)
         d_map[i] = map[i];
 }
 
 std::vector<int> map_bb_impl::map() const
 {
-    std::vector<int> m;
-    for (unsigned i = 0; i < 0x100; i++)
+    gr::thread::scoped_lock guard(d_mutex);
```

Comment:
Technically I don't think this guard is required, since it's not changing anything internal to the class. That said, I prefer using it in places like this to guarantee that the output result is valid, since it is technically possible that one thread could be doing `::map` while another is doing ``::set_map` ... and this protects against that possibility (not matter how remote it is).